packages/babel-generator/src/buffer.ts

```diff
@@ -237,9 +255,26 @@ export default class Buffer {
       return;
     }
 
-    const { column, identifierName, filename } = sourcePos;
+    let { column, identifierName, identifierNamePos, filename } = sourcePos;
     let line = sourcePos.line;
 
+    if (identifierName != null || identifierNamePos != null) {
+      sourcePos.identifierName = undefined;
+      sourcePos.identifierNamePos = undefined;
+    }
+
+    // Fast path for multi-line
+    if (maybeNewline && identifierNamePos && this._map?._inputMap) {
```

Comment:
What's this block used for?

Reply:
We'll call `_mark` on each line, so we can avoid having to look up the original identifier name multiple times, maybe always search here?

Reply:
Oh, we shouldn't mark a name for the start of a line. We can just pass nulls for the while loop